### ISSUE TITLE
Feature Request: Sample check

### ISSUE THREAD

**Issue by faltaren** [airdcpp-web/airdcpp-release-validator#12]
I would like to have a sample checker. Samples are often broken. 

I see two different way to search:

Check with srrdb.com and rebuild the sample it it doesn´t match. 

If the rebuild doesn´t work or its not on srrdb do a search and download the biggest/match srrdb sample with the same name. 
m2ts cant be rebuilt so the search is needed here.

**Comment by maksis** [comment]
If the extension would connect to srrdb or similar services, I believe that it could also validate other files, as there are often smaller files with different sizes being shared (e.g. NFOs, random mismatching image files). Having identical content for the shared releases would be optimal.

As this extension currently works only based on the content on disk, I'm not sure whether it would be more suitable to implement a separate srrdb validator extension, as the idea of using an external service is quite different from the current approach (and it would increase the scope of the extension significantly). Some of the code from this extension could be used as a basis for that though.

**Comment by faltaren** [comment]
I agree. I would be omptimal to check all the files. Add a seperate addon would work to. I cant help with coding but I can help with testing if anyone is up for the building of the extension.

**Comment by peps1** [comment]
mkv sample validation could be done locally using mkvinfo. 
example shell script: https://github.com/peps1/titto-glftpd/blob/master/mkv_check/mkv_check.sh#L112

**Comment by faltaren** [comment]
Ok thats nice to find them. How about m2ts?

**Comment by peps1** [comment]
m2ts is pretty special, they can sometimes be restored, though I don't think it ever worked for me. But I think there is no way to find out if an m2ts is actually complete like you can with mkv.

**Comment by faltaren** [comment]
But if the addon checks with srrdb and then search the bubs for the closest match?

**Comment by gangefors** [comment]
There is one important consideration if a plugin should start using srrdb and it is that they only allow a limited number of daily requests from the same IP. Don't give them a reason to limit that further or end up not being able to pay their bandwidth bills. 😉
